### PR TITLE
feat(button): new component specs

### PR DESCRIPTION
**Pull request by BY00565233**
## Description

`<hx-button>` : Implementing color palette styles

### What are the relevant story cards/tickets? Any additional PRs or other references?

Jira: SURF-2152

## Before you request a review for this PR:

- [x] For UI changes, did you manually test in recent versions of modern browsers (Chrome, Firefox, and Safari)?
- [x] For UI changes, did you manually test in IE11 and legacy Edge?
- [ ] Did you add component tests for any new code?
- [ ] Did you run the component unit tests via `yarn test` to ensure all tests passed?
- [ ] Did you include a screenshot of the component tests?
- [ ] If you changed/added functionality, did you update the demo page and documentation?
- [ ] If needed, did you add or modify the demo test page to test the changed/added functionality?
- [x] Did you assign reviewers?
- [x] In Jira, have you linked to this PR on the ticket(s)?

### CODE REVIEW

File: src/scss/components/button/mixins/_hxButton.scss [hunk 1]
```diff
@@ -86,7 +86,7 @@
 /// ------------------------------------ ///
 
 @mixin hxButton--primary {
-  background-color: $cyan-900;
+  background-color: $cyan-500;
```
Comment (100stacks) [inline src/scss/components/button/mixins/_hxButton.scss:89]: Please update to the new `v2.0` color variables:

https://github.com/HelixDesignSystem/helix-ui/blob/eb547a4294af4f768294ffef085cfa58136718c1/src/scss/_vars.scss#L1-L118

File: src/scss/components/button/mixins/_hxButton.scss [hunk 2]
```diff
@@ -107,15 +107,15 @@
 }
 
 @mixin hxButton-active--primary {
-  background-color: $cyan-700;
+  background-color: $cyan-900;
```
Comment (100stacks) [inline src/scss/components/button/mixins/_hxButton.scss:110]: Please update to the new `v2.0` color variables:

https://github.com/HelixDesignSystem/helix-ui/blob/eb547a4294af4f768294ffef085cfa58136718c1/src/scss/_vars.scss#L1-L118

File: src/scss/components/button/mixins/_hxButton.scss [hunk 3]
```diff
@@ -124,7 +124,7 @@
 }
 
 @mixin hxButton-hover--primary {
-  background-color: $cyan-500;
+  background-color: $cyan-700;
```
Comment (100stacks) [inline src/scss/components/button/mixins/_hxButton.scss:127]: Please update to the new `v2.0` color variables:

https://github.com/HelixDesignSystem/helix-ui/blob/eb547a4294af4f768294ffef085cfa58136718c1/src/scss/_vars.scss#L1-L118

File: src/scss/components/button/mixins/_hxButton.scss [hunk 4]
```diff
@@ -155,23 +155,23 @@
 }
 
 @mixin hxButton-active--secondary {
-  background-color: $cyan-700;
+  background-color: $cyan-900;
```
Comment (100stacks) [inline src/scss/components/button/mixins/_hxButton.scss:158]: Please update to the new `v2.0` color variables:

https://github.com/HelixDesignSystem/helix-ui/blob/eb547a4294af4f768294ffef085cfa58136718c1/src/scss/_vars.scss#L1-L118

File: src/scss/components/button/mixins/_hxButton.scss [hunk 5]
```diff
@@ -155,23 +155,23 @@
...
+  color: $gray-500;
 }
 
 @mixin hxButton-focus--secondary {
   box-shadow: $focus-glow;
 }
 
 @mixin hxButton-hover--secondary {
-  background-color: $cyan-500;
+  background-color: $cyan-700;
```
Comment (100stacks) [inline src/scss/components/button/mixins/_hxButton.scss:174]: Please update to the new `v2.0` color variables:

https://github.com/HelixDesignSystem/helix-ui/blob/eb547a4294af4f768294ffef085cfa58136718c1/src/scss/_vars.scss#L1-L118

File: src/scss/components/button/mixins/_hxButton.scss [hunk 6]
```diff
@@ -184,7 +184,7 @@
 @mixin hxButton--tertiary {
   background-color: transparent;
   border: 0;
-  color: $cyan-900;
+  color: $cyan-500;
```
Comment (100stacks) [inline src/scss/components/button/mixins/_hxButton.scss:187]: Please update to the new `v2.0` color variables:

https://github.com/HelixDesignSystem/helix-ui/blob/eb547a4294af4f768294ffef085cfa58136718c1/src/scss/_vars.scss#L1-L118

File: src/scss/components/button/mixins/_hxButton.scss [hunk 7]
```diff
@@ -212,13 +212,13 @@
 @mixin hxButton-active--tertiary {
   background-color: transparent;
   border-color: transparent;
-  color: $cyan-700;
+  color: $cyan-900;
```
Comment (100stacks) [inline src/scss/components/button/mixins/_hxButton.scss:215]: Please update to the new `v2.0` color variables:

https://github.com/HelixDesignSystem/helix-ui/blob/eb547a4294af4f768294ffef085cfa58136718c1/src/scss/_vars.scss#L1-L118

File: src/scss/components/button/mixins/_hxButton.scss [hunk 8]
```diff
@@ -228,5 +228,5 @@
 @mixin hxButton-hover--tertiary {
   background-color: transparent;
   border-color: transparent;
-  color: $cyan-500;
+  color: $cyan-700;
```
Comment (100stacks) [inline src/scss/components/button/mixins/_hxButton.scss:231]: Please update to the new `v2.0` color variables:

https://github.com/HelixDesignSystem/helix-ui/blob/eb547a4294af4f768294ffef085cfa58136718c1/src/scss/_vars.scss#L1-L118

File: src/scss/components/typography/_index.scss [hunk 9]
```diff
@@ -64,9 +64,9 @@ hx-error {
 // Additional Helper Classes
 .hxSubdued {
   color: $gray-700;
-  font-size: 0.875rem;
+  font-size: 0.625rem;
   font-weight: 400;
-  line-height: 1.5rem;
+  line-height: 1;
```
Comment (100stacks) [inline src/scss/components/typography/_index.scss:69]: We should make these changes specific to the Pill component.

**Review by 100stacks — CHANGES_REQUESTED** [review]

File: src/scss/components/button/mixins/_hxButton.scss (hunk 1 above)
Comment (BY00565233) [reply src/scss/components/button/mixins/_hxButton.scss]: Changes have been done, Please review

File: src/scss/components/button/mixins/_hxButton.scss (hunk 2 above)
Comment (BY00565233) [reply src/scss/components/button/mixins/_hxButton.scss]: Changes have been done, Please review

File: src/scss/components/button/mixins/_hxButton.scss (hunk 3 above)
Comment (BY00565233) [reply src/scss/components/button/mixins/_hxButton.scss]: Changes have been done, Please review

File: src/scss/components/button/mixins/_hxButton.scss (hunk 4 above)
Comment (BY00565233) [reply src/scss/components/button/mixins/_hxButton.scss]: Changes have been done, Please review

File: src/scss/components/button/mixins/_hxButton.scss (hunk 5 above)
Comment (BY00565233) [reply src/scss/components/button/mixins/_hxButton.scss]: Changes have been done, Please review

File: src/scss/components/button/mixins/_hxButton.scss (hunk 6 above)
Comment (BY00565233) [reply src/scss/components/button/mixins/_hxButton.scss]: Changes have been done, Please review

File: src/scss/components/button/mixins/_hxButton.scss (hunk 7 above)
Comment (BY00565233) [reply src/scss/components/button/mixins/_hxButton.scss]: Changes have been done, Please review

File: src/scss/components/button/mixins/_hxButton.scss (hunk 8 above)
Comment (BY00565233) [reply src/scss/components/button/mixins/_hxButton.scss]: Changes have been done, Please review

File: src/scss/components/typography/_index.scss (hunk 9 above)
Comment (BY00565233) [reply src/scss/components/typography/_index.scss]: Changes have been done, Please review

**Review by danielmdesigns — CHANGES_REQUESTED** [review]
There is a darker blue border on the primary buttons. Please remove this border, there should be no border on primary buttons.

The secondary button should not have a fill on hover state. Should only be outlined. The color changes should be as follows:

Default: Blue-500
Hover: Blue-700
Pressed: Blue-900

File: src/scss/components/pill/_index.scss [hunk 10]
```diff
@@ -13,8 +13,6 @@ hx-status {
   color: $gray-750;
   display: inline-block;
   font-size: 0.625rem; // 10px
-  height: 1rem;
-  line-height: 1;
```
Comment (100stacks) [inline src/scss/components/pill/_index.scss:17]: ### Please revert these changes.  I'll add the patch to PR #822.

**Review by 100stacks — CHANGES_REQUESTED** [review]

**Review by 100stacks — COMMENTED** [review]
### @danielmdesigns, please review commit 368013da5200cd7474de267db8372548e1e4889c updates to this PR #821:


#### https://deploy-preview-821--helix-ui.netlify.app/components/button

**Review by danielmdesigns — APPROVED** [review]
Buttons have been corrected and look good to me. Approved.

**Review by lalithkarikelli — APPROVED** [review]
DEV LGTM

**Review by 100stacks — APPROVED** [review]
Dev LGTM

**Pull request merged**